### ISSUE TITLE
cgo_c_lib must produce a single file error

### ISSUE THREAD

**Issue by ivyxyz** [bazelbuild/rules_go#896]
Testing e254d73bf1181101ed82791fa5d204c4c5b1b105 with a larger repository, this error is shown. When I make a small repository with one go file linking org_golang_x_sys for testing, this error is not triggered. This seems to be caused by 567738b9d3b61b162c1eda5f7ed21751379122d6.

`ERROR: external/org_golang_x_sys/unix/BUILD.bazel:3:1: in lib attribute of _cgo_collect_info rule @org_golang_x_sys//unix:go_default_library.cgo_info: '@org_golang_x_sys//unix:go_default_library.cgo_c_lib' must produce a single file. Since this rule was created by the macro 'go_library_macro', the error might have been caused by the macro implementation in external/io_bazel_rules_go/go/private/rules/cgo.bzl:314:13`

**Comment by ianthehat** [comment]
The `.cgo_c_lib` target is a `native.cc_library`
I think it always has at least one output, so I am guessing there is something that is triggering it to produce more than one file, not sure what those conditions would be. I don't think the size of the repository could have any effect, it could be a change in the cc toolchain or any of the options to the cc compiler that the other repository embeds though.

It would be interesting if you could build `@org_golang_x_sys//unix:go_default_library.cgo_c_lib` directly and tell us what files it says it generates (in the failing repository)

Either way, the `_cgo_collect_info` rule probably needs to be adjusted to pick out the archive output of it's `lib` attribute rather than assume that `cc_library` only has one file output that is the archive.

**Comment by ivyxyz** [comment]
Thanks for the hint, I've found out the cause. Building with `-c opt` will trigger the error.

```
bazel build @org_golang_x_sys//unix:go_default_library.cgo_c_lib
INFO: Found 1 target...
Target @org_golang_x_sys//unix:go_default_library.cgo_c_lib up-to-date:
  bazel-bin/external/org_golang_x_sys/unix/libgo_default_library.cgo_c_lib.lo
INFO: Elapsed time: 0.351s, Critical Path: 0.00s

bazel build -c opt @org_golang_x_sys//unix:go_default_library.cgo_c_lib
INFO: Found 1 target...
Target @org_golang_x_sys//unix:go_default_library.cgo_c_lib up-to-date:
  bazel-bin/external/org_golang_x_sys/unix/libgo_default_library.cgo_c_lib.lo
  bazel-bin/external/org_golang_x_sys/unix/libgo_default_library.cgo_c_lib.pic.lo
INFO: Elapsed time: 1.095s, Critical Path: 0.25s
```

**Comment by rlguarino** [comment]
I just ran into the same issue myself. For me it was using a `go_binary` as a tar input for a `container_image` that was used in as a image in a `k8s_object` rule that triggered the issue. I can confirm I have the same output as @ivyxyz.

```
$ bazel build -c opt @org_golang_x_sys//unix:go_default_library.cgo_c_lib
INFO: Analysed target @org_golang_x_sys//unix:go_default_library.cgo_c_lib (0 packages loaded).
INFO: Found 1 target...
Target @org_golang_x_sys//unix:go_default_library.cgo_c_lib up-to-date:
  bazel-bin/external/org_golang_x_sys/unix/libgo_default_library.cgo_c_lib.lo
  bazel-bin/external/org_golang_x_sys/unix/libgo_default_library.cgo_c_lib.pic.lo
INFO: Elapsed time: 0.351s, Critical Path: 0.03s
INFO: Build completed successfully, 1 total action
$ bazel build @org_golang_x_sys//unix:go_default_library.cgo_c_lib
INFO: Analysed target @org_golang_x_sys//unix:go_default_library.cgo_c_lib (0 packages loaded).
INFO: Found 1 target...
Target @org_golang_x_sys//unix:go_default_library.cgo_c_lib up-to-date:
  bazel-bin/external/org_golang_x_sys/unix/libgo_default_library.cgo_c_lib.lo
INFO: Elapsed time: 0.791s, Critical Path: 0.04s
INFO: Build completed successfully, 1 total action
```

**Comment by razvanm** [comment]
I also just bump into this.

Happy build:

```
$ bazel build //vendor/github.com/mattn/go-sqlite3/...
.........
INFO: Analysed 9 targets (16 packages loaded).
INFO: Found 9 targets...
INFO: Elapsed time: 8.907s, Critical Path: 4.92s
INFO: Build completed successfully, 25 total actions
$
```

Sad build using `-c opt`:

```
$ bazel build -c opt //vendor/github.com/mattn/go-sqlite3/...
ERROR: /home/razvanm/work/grail/go/src/grail.com/vendor/github.com/mattn/go-sqlite3/BUILD:5:1: in lib attribute of _cgo_collect_info rule //vendor/github.com/mattn/go-sqlite3:go_default_library.cgo_info: '//vendor/github.com/mattn/go-sqlite3:go_default_library.cgo_c_lib' must produce a single file. Since this rule was created by the macro 'go_library_macro', the error might have been caused by the macro implementation in /home/razvanm/.cache/bazel/_bazel_razvanm/5f5cf307ace2d67d2365ceb10c844a87/external/io_bazel_rules_go/go/private/rules/cgo.bzl:314:13
ERROR: Analysis of target '//vendor/github.com/mattn/go-sqlite3:go_default_library.cgo_info' failed; build aborted
INFO: Elapsed time: 0.458s
FAILED: Build did NOT complete successfully (0 packages loaded)
$ 
```
